src/platform_impl/linux/x11/window.rs

```diff
@@ -424,6 +424,21 @@ impl UnownedWindow {
                 );
             } //.queue();
 
+            // Set the transient parent if we don't have a parent already.
+            if event_loop.root == root {
+                let owner = pl_attribs.owner.map(|owner| match owner {
+                    RawWindowHandle::Xlib(x) => x.window,
+                    RawWindowHandle::Xcb(x) => x.window as u64,
+                    raw => unreachable!("Invalid raw window handle {raw:?} on X11"),
+                });
+
+                if let Some(owner) = owner {
+                    unsafe {
+                        (xconn.xlib.XSetTransientForHint)(xconn.display, window.xwindow, owner);
```

Comment:
As I understand, for right click menus, drop downs, etc. `override_redirect` Windows are typically used on X11. This is what `GdkPopover` uses. GDK separately has a method that sets `WM_TRANSIENT_FOR`: https://docs.gtk.org/gdk4/method.Toplevel.set_transient_for.html. On Wayland that corresponds to https://wayland.app/protocols/xdg-shell#xdg_toplevel:request:set_parent.

> Setting the override_redirect attribute True for a window allows a window to be mapped, moved, resized, or its border width changed without the intervention of the window manager. This override is usually done for menus that are frequently mapped and almost immediately unmapped again.
>
> Under properly designed window managers, there is a property you can set to tell the window manager to allow a window to pop up with minimal intervention (XA_WM_TRANSIENT_FOR). This is used for dialog boxes, as described in [12.3.1.4.6 Transient Window Field](https://www.oreilly.com/library/view/xlib-programming-manual/9780596806187/ch12s03.html#transient_window_field).

https://www.oreilly.com/library/view/xlib-programming-manual/9780596806187/ch04s03.html

So in Wayland and X, a "popup" and a "transient toplevel" are different things.

Reply:
Thanks for telling me! I'll rework this PR once #2614 lands and I can more easily mold the X11 backend into something that is more conducive to this